### PR TITLE
9264 add analyses events

### PR DESCRIPTION
**Pull request by gfiorav**
This adds events for analyses.

Please CR @juanignaciosl !

### CODE REVIEW

File: lib/carto/tracking/events.rb [hunk 1]
```diff
@@ -175,6 +175,33 @@ class LikedMap < Event
 
         required_properties :user_id, :visualization_id, :action
       end
+
+      class CreatedAnalysis < Event
```
Comment (juanignaciosl) [inline lib/carto/tracking/events.rb]: What about an `AnalysisEvent` including those three mixins and `required properties`, making the three specific classes three one-liners?

**Comment by juanignaciosl** [comment]
👍 after considering `AnalysisEvent` class.

File: lib/carto/tracking/events.rb (hunk 1 above)
Comment (gfiorav) [inline lib/carto/tracking/events.rb]: The idea was to keep it as easy to add new events as possible. Adding a mixin would make it a tiny bit less straightforward to understand. Still could do it.

Did I convince you? If not I'll change it for sure.

Comment (juanignaciosl) [inline lib/carto/tracking/events.rb]: On the contrary, I'm not suggesting a mixing but a class:

```ruby
class AnalysisEvent < Event
  include Carto::Tracking::Services::Segment
 
  include Carto::Tracking::Validators::Visualization::Writable
  include Carto::Tracking::Validators::User

  required_properties :user_id, :visualization_id, :analysis
end

class CreatedAnalysis < AnalysisEvent ; end
class ModifiedAnalysis < AnalysisEvent ; end
class DeletedAnalysis < AnalysisEvent ; end
```

I think that this is simpler and this inheritance makes sense (all three events _are_ analysis events).

**Comment by gfiorav** [comment]
@juanignaciosl it was harder than expected. found a descendant proof way to maintain `required_properties` scheme.

Have a second look please!

**Comment by juanignaciosl** [comment]
Nice, complete refactor! 👌

**Pull request merged**